### PR TITLE
Adds examples, closes #2

### LINKED ISSUE #2: README needs simple usage example
- [ ] Create a list
- [ ] Add elements
- [ ] Iterate through the list

### PR DESCRIPTION
**Pull request by jpstroop**
If I understand the API correctly, this should do it for now.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,6 +11,71 @@ prescribed by the [Hydra Common Data Model](https://wiki.duraspace.org/display/h
 This work is the result of meetings in Nov. 2014 and Feb. 2015 at the White
 Stag building in Portland, OR.
 
+## Sample Usage
+
+### Create an Empty Proxy List
+```ruby
+ > require 'rdf/proxy_list'
+ > my_aggregator = RDF::URI('http://example.org')/'my_agg'
```
Comment (jcoyne) [inline README.md]: This looks like a SyntaxError

Comment (jcoyne) [inline README.md]: And I guess I'm wrong. But why not `RDF::URI('http://example.org/my_agg')` instead?

Comment (jpstroop) [inline README.md]:
``` ruby
2.1.4 :001 > require 'rdf'
 => true 
2.1.4 :002 > RDF::URI('http://example.org')/'my_agg'
 => #<RDF::URI:0x8b8594 URI:http://example.org/my_agg>
```

Comment (jpstroop) [inline README.md]: I had 'http://example.org' as a constant...lazy find-and-replacing ensued... You're right, I should just do `RDF::URI('http://example.org/my_agg')`. Hold on.

Comment (jcoyne) [inline README.md]: Thanks. That makes it much easier to focus on the intent rather than the unusual looking syntax. :+1:

**Comment by jcoyne** [comment]
Perfect! Thanks.

**Pull request merged**